sdks/python/apache_beam/runners/common.py

```diff
@@ -1035,6 +1284,8 @@ def process_outputs(self, windowed_input_element, results):
           windowed_value.windows *= len(windowed_input_element.windows)
       else:
         windowed_value = windowed_input_element.with_value(result)
+      if watermark_estimator:
```

Comment:
`is not None` (this is performance-sensitive code, as it's called for every value for every transform).

Reply:
Thanks for pointing out the performance issue!

Reply:
The checking is needed for non-sdf dofn since we only have 'watermark_estimator` in SDF.